### PR TITLE
Don't mark SitePage node as dirty if node data didn't change

### PR DESCRIPTION
**Pull request by pieh**
Right now we would mark `SitePage` as dirty because we generate `contentDigest` using `updatedAt` field which isn't passed to `SitePage` node.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit 1b37dbf48d0623589e7cdff629d2bd2717fbce41

https://app.netlify.com/sites/using-postcss-sass/deploys/5b55b72a4ed62f1c02e8c3ec

File: packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js [hunk 1]
```diff
@@ -74,7 +74,12 @@ exports.runQueuedActions = runQueuedActions
 // Wait until all plugins have finished running (e.g. various
 // transformer plugins) before running queries so we don't
 // query things in a 1/2 finished state.
-emitter.on(`API_RUNNING_QUEUE_EMPTY`, runQueuedActions)
+emitter.on(`API_RUNNING_QUEUE_EMPTY`, async api => {
```
Comment (KyleAMathews) [inline packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js]: Is this reliable? Does calling `createNode` always finish with an `onCreateNode` API call? E.g. in the future we might be encouraging people to create pages in response to nodes being created in which case the last API call would be `onCreatePage`.

**Comment by KyleAMathews** [comment]
We do merge queries in the query-queue https://github.com/gatsbyjs/gatsby/blob/193dbb1de76068eb25518f4a0715141eccbf70bd/packages/gatsby/src/internal-plugins/query-runner/query-queue.js#L12-L14

Is that not working? Or is there times when a query finishes and then is triggered afterwards?

**Comment by freiksenet** [comment]
@KyleAMathews Seems like it is triggered twice, I've logged at apollo-link level so it was after all gatsby stuff.

File: packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js (hunk 1 above)
Comment (pieh) [reply packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js]: That's good question. I'll try creating pages in `onCreateNode` and report back.

**Comment by pieh** [comment]
I reverted part that potentially could introduce more bugs than it fixes issues, leaving only small change that doesn't mark `SitePage` nodes as dirty when only `updatedAt` part of page changed (which isn't added to `SitePage` node anyway)

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit 5b6dbf43e5b3ae7f0424a1828e58f7d1835bc295

https://app.netlify.com/sites/gatsbyjs/deploys/5b59eb1fc6aed65f7505d4a8

**Pull request merged**